### ISSUE TITLE
"Entity Not Available"

### ISSUE THREAD

**Issue by itm1960** [fishbigger/HomeAssistant-Tapo-P100-Control#46]
I have three Tapo P100s, and have had all three working well with Home Assistant Core (v2021.10.3) for a few months, thanks to this excellent component. 
Yesterday however, two of the three plugs became inaccessible to HA - there's an "Entity Not Available" message alongside them on my Lovelace dashboard. The third is still accessible. Both of the problem plugs are still accessible and controllable by the Tapo app on my phone, and their IP addresses remain correct (i.e. the same as those referenced in my configuration.yaml). 
I've tried restarting HA twice, but the same two plugs remain unavailable.
Has anyone else had this problem?

**Comment by itm1960** [comment]
Edit: I've just noticed that the P100s that are still working are running firmware v1.3.4, whereas the two that have dropped offline (from HA) are running v1.4.9. 
???

**Comment by chsims1** [comment]
I've just tried a v1.4.9 today for the first time, and the entity is not recognised.

**Comment by JImHaywood90** [comment]
I have managed to get my switches running 1.4.9 imported into Home Assistant via homebridge. Not ideal but at least it’s working. [https://github.com/apatsufas/homebridge-tapo-p100#readme]

**Comment by fishbigger** [comment]
Hi, can you post your logs related to the entity? To help debug it.

**Comment by wardy277** [comment]
> Hi, can you post your logs related to the entity? To help debug it.

I am also having this issue. Logs are:

```
Logger: homeassistant.components.switch
Source: custom_components/tapo_p100_control/switch.py:86
Integration: Switch (documentation, issues)
First occurred: 16:46:58 (1 occurrences)
Last logged: 16:46:58

Error while setting up tapo_p100_control platform for switch
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/tapo_p100_control/switch.py", line 44, in setup_platform
    add_entities([P100Plug(p100)])
  File "/config/custom_components/tapo_p100_control/switch.py", line 53, in __init__
    self.update()
  File "/config/custom_components/tapo_p100_control/switch.py", line 86, in update
    data = self._p100.getDeviceInfo()
  File "/usr/local/lib/python3.9/site-packages/PyP100/PyP100.py", line 252, in getDeviceInfo
    decryptedResponse = self.tpLinkCipher.decrypt(r.json()["result"]["response"])
KeyError: 'result'
```

I have updated the custom component to the latest from this github and am on the latest homeassistant 2021.12.9 and restarted

**Comment by stevendodd** [comment]
I am also experiencing this problem

**Comment by AdamHughes** [comment]
I am experiencing the same problem. One plug is available, whereas the other is not. Both are on firmware 1.2.x (I have not let them update since they arrived) 

My HA config is:
`
switch roomfan:
    platform: tapo_p100_control
    ip_address: 192.168.0.54
    email: xxxxxxxxxxx
    password: xxxxx
switch shelflamp:
    platform: tapo_p100_control
    ip_address: 192.168.0.55
    email: xxxxxxx
    password: xxxxxxxx

`

Any ideas?

**Comment by stevendodd** [comment]
> I am also experiencing this problem, I think it's related to multiple switches. I have it working currently with the following configuration.
> 
> ```
> switch:
>     - platform: tapo_p100_control
>       ip_address: 192.168.X.X
>       email: user@gmail.com
>       password: Password
>     - platform: tapo_p100_control
>       ip_address: 192.168.X.X
>       email: user@gmail.com
>       password: Password
> ```

I should clarify, I have one plug with updated firmware and the other not. In addition to this my setup is working at the moment and has been quite stable since my last post, I had previously had different settings but the settings listed above once everything had been deleted and then re-added seem to be working fine.

**Comment by AdamHughes** [comment]
How odd lol, it is working perfectly now, Thank you!!

**Comment by itm1960** [comment]
My problem has just returned. I have 4 P100s, all of which were working fine yesterday. Today, two of them are offline:
- #1 - firmware v1.3.4 - online
- #2 - firmware v1.4.9 - online
- #3 - firmware v1.4.9 - Entity not available
- #2 - firmware v1.4.9 - Entity not available.
It all seems pretty random :0(

**Comment by AdamHughes** [comment]
Yeah, experienced the same today. It seems that removing them from the home assistant config, restarting HA and then adding them back and restarting HA fixes it, odd.

**Comment by itm1960** [comment]
Thanks - that seemed to do the trick, Well, sort of.....now one of the other P100s is showing as "Device unreachable" in the Tapo app. But I guess that's not a Home Assistant problem (I don't have that device configured in HA).

I also found that one of the other P100s (one of those which was unavailable in HA yesterday) became "unreachable" in the Tapo app yesterday after I unplugged it and re-inserted it (a few times). It's now back online (without intervention since it became unreachable yesterday), but perhaps this indicates problems with the reliability of these plugs outside of the HA environment?
